src/task_system/task_system_server/Controllers/TaskSystemController.cs

```diff
@@ -26,37 +26,41 @@ public TaskSystemController(ITaskSystemRepository taskSystemRepository, IPermiss
         [HttpGet]
         public async Task<IActionResult> GetTasks([FromQuery] QueryObject query)
         {
-            if (!Request.Headers.TryGetValue("X-API-Key", out var apiKey) || !_permissionValidator.ValidatePermissions(PermissionNames.VIEW_TASKS, apiKey!))
+            if (!Request.Headers.TryGetValue("X-API-Key", out var apiKey))
                 return Unauthorized(new TaskSystemResponse<string>(ResponseMessages.UNAUTHORIZED, null));
 
+            if(!_permissionValidator.ValidatePermissions(PermissionNames.VIEW_TASKS, apiKey) && 
+                (!query.AccountId.HasValue || !_permissionValidator.ValidateAccountId(query.AccountId.Value, apiKey!) || !_permissionValidator.ValidatePermissions(PermissionNames.CREATE_TASKS, apiKey!)))
```

Comment:
## Non-Blocking

### 1) Comment here would be useful
- This is a very complicated if statement, it would be nice to have something like "Authorized if either the user can ViewTasks, or if they can CreateTasks and they are filtering on their own accountId"

### 2) Unexpected Unauthorized response
- Currently, the front end must know if the user can ViewTasks or only their own tasks. Ie, GET /tasks returns unauthorized if the user can only CreateTasks, whereas that same user gets a response on /tasks?AccountId=1, this is not intuitive from a frontend dev PoV. They should be able to simple GET /tasks and the server should figure out which tasks they can see. 
### Solution
```
if not validatePermissions(VIEW_TASKS): 
    query.AccountId = _permissionClient.GetSessionData(sessionKey)
```
- Note, getting the accountId from the session key can be completed in 1 of 2 ways:
1) include the _permissionClient in the controller constructor. This would require a change to Program.cs:line61, since we would need a `builder.Services.AddScoped<MockPermissionClient>()` in the "else" case (including a new mock permission client)
2) modify the permissionsValidator to include some sort way of returning the permData object. The best way to do that would be to include an "out" parameter in that function by overloading the methods with :
``` public bool ValidatePermissions(string permissionName, string sessionKey, out SessionData paramData) ```
and
``` public bool ValidateAccountId(int accountId, string sessionKey, out SessionData paramData) ```